plugins/processors/awsapplicationsignals/internal/resolver/attributesresolver.go

```diff
@@ -0,0 +1,161 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package resolver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	semconv "go.opentelemetry.io/collector/semconv/v1.22.0"
+	"go.uber.org/zap"
+
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/common"
+	appsignalsconfig "github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/config"
+	attr "github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsapplicationsignals/internal/attributes"
+	"github.com/aws/amazon-cloudwatch-agent/translator/util/ecsutil"
+)
+
+const (
+	AttributeEnvironmentDefault = "default"
+
+	AttributePlatformGeneric = "Generic"
+	AttributePlatformEC2     = "AWS::EC2"
+	AttributePlatformEKS     = "AWS::EKS"
+	AttributePlatformK8S     = "K8s"
+)
+
+var GenericInheritedAttributes = map[string]string{
+	semconv.AttributeDeploymentEnvironment: attr.AWSLocalEnvironment,
+	attr.ResourceDetectionHostName:         common.AttributeHost,
+}
+
+// DefaultInheritedAttributes is an allow-list that also renames attributes from the resource detection processor
+var DefaultInheritedAttributes = map[string]string{
+	semconv.AttributeDeploymentEnvironment: attr.AWSLocalEnvironment,
+	attr.ResourceDetectionASG:              common.AttributeEC2AutoScalingGroup,
+	attr.ResourceDetectionHostId:           common.AttributeEC2InstanceId,
+	attr.ResourceDetectionHostName:         common.AttributeHost,
+}
+
+type subResolver interface {
+	Process(attributes, resourceAttributes pcommon.Map) error
+	Stop(ctx context.Context) error
+}
+
+type attributesResolver struct {
+	subResolvers []subResolver
+}
+
+// create a new attributes resolver
+func NewAttributesResolver(resolvers []appsignalsconfig.Resolver, logger *zap.Logger) *attributesResolver {
+	subResolvers := []subResolver{}
+	for _, resolver := range resolvers {
+		switch resolver.Platform {
+		case appsignalsconfig.PlatformEKS, appsignalsconfig.PlatformK8s:
+			subResolvers = append(subResolvers, getKubernetesResolver(resolver.Platform, resolver.Name, logger), newKubernetesResourceAttributesResolver(resolver.Platform, resolver.Name))
+		case appsignalsconfig.PlatformEC2:
+			subResolvers = append(subResolvers, newResourceAttributesResolver(resolver.Platform, AttributePlatformEC2, DefaultInheritedAttributes))
+		default:
+			if ecsutil.GetECSUtilSingleton().IsECS() {
```

Comment:
Note: This is the main change. I am cross referencing this file with https://github.com/aws/amazon-cloudwatch-agent/blob/main/plugins/processors/awsappsignals/internal/resolver/attributesresolver.go